tests/sentence_transformers/test_sentence_transformers_model_export.py

```diff
@@ -285,3 +304,98 @@ def test_default_signature_assignment(model_path, basic_model):
     default_signature = mlflow.sentence_transformers._get_default_signature()
 
     assert default_signature.to_dict() == expected_signature
+
+
+def test_model_pyfunc_save_load(basic_model, model_path):
+    mlflow.sentence_transformers.save_model(basic_model, model_path)
+    loaded_pyfunc = pyfunc.load_model(model_uri=model_path)
+
+    sentence = "hello world and hello mlflow"
+    sentences = [sentence, "goodbye my friends", "i am a sentence"]
+    embedding_dim = basic_model.get_sentence_embedding_dimension()
+
+    emb0 = loaded_pyfunc.predict(sentence)
+    assert emb0.shape == (1, 1)
+    assert len(emb0.iloc[0,0]) == embedding_dim
+
+    emb1 = loaded_pyfunc.predict(sentences)
+    emb2 = loaded_pyfunc.predict(pd.Series(sentences))
+    emb3 = loaded_pyfunc.predict(pd.Series(sentences).to_numpy())
+
+    for emb in [emb1, emb2, emb3]:
+        assert emb.shape == (3, 1)
+        assert len(emb.iloc[0,0]) == embedding_dim
+        assert len(emb.iloc[1,0]) == embedding_dim
+        assert len(emb.iloc[2,0]) == embedding_dim
+
+    pd.testing.assert_frame_equal(emb1, emb2)
+    pd.testing.assert_frame_equal(emb1, emb3)
+
+
+def test_spark_udf(basic_model, model_path, spark):
+    with mlflow.start_run():
+        model_info = mlflow.sentence_transformers.log_model(basic_model, "my_model")
+    loaded_model = mlflow.pyfunc.spark_udf(
+        spark, model_info.model_uri, result_type="string", # result_type=ArrayType(FloatType())
+    )
+
+    df = spark.createDataFrame([("hello MLflow",), ("bye world",)], ["text"])
+    df = df.withColumn("embedding", loaded_model("text"))
+    pdf = df.toPandas()
+
+    emb_string = pdf.loc[0, "embedding"]
+    emb = ast.literal_eval(emb_string)
+    assert len(emb) == basic_model.get_sentence_embedding_dimension()
```

Comment:
@BenWilson2 Seems like `spark_udf` output an array object. The docs indicate that `result_type=ArrayType(...)` indicates multiple columns of that type, rather than a single column with array contents.

Do you know if there's a better way to do this? Or will users also have to apply the spark_udf to get a string result, and parse the string back into a list?